### ISSUE TITLE
Rest/sphinx filter when spellchecking comments

### ISSUE THREAD

**Issue by mitar** [PyCQA/pylint#1587]
### Steps to reproduce
1. Have a comment like:
```
# Should we use ``len()`` or ``length()``?
```
2. Enable spell-checking of comments by installing `pyenchant` and choosing en_US as the language
3. Run pylint.

### Current behavior

I get the error:

```
Did you mean: ''Len' or 'en' or 'glen' or 'Olen''? (wrong-spelling-in-comment)
```

### Expected behavior

I do not get any error.

I think spellchecking should remove rest/sphinx syntax before running spellchecking. Probably a filter like it is currently done for links and wiki syntax should be added. See https://github.com/rfk/pyenchant/issues/113.

### pylint --version output

```
pylint 1.7.2, 
astroid 1.5.3
Python 3.5.2 (default, Oct 11 2016, 05:00:16) 
[GCC 4.2.1 Compatible Apple LLVM 7.0.2 (clang-700.1.81)]
```

**Comment by PCManticore** [comment]
Sounds good.

**Comment by phi-dbq** [comment]
It will be nice to also check reference with Sphinx domain, e.g. `py:class:XXX`.

**Comment by Pierre-Sassoulas** [comment]
Closing as it does not raise anymore on 2.14.

**Comment by Pierre-Sassoulas** [comment]
Yes, I think the issue was about the rst syntax around the ``len``.
